### PR TITLE
Deliver multiple KeyVault support

### PR DESCRIPTION
**Pull request by mnriem**
## Summary

Delivers support for multiple keyvaults

## Issue Type
- New Feature

## Starter Names
  - key vault spring boot starter

## Additional Information

### CODE REVIEW

File: azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java [hunk 1]
```diff
@@ -46,16 +45,28 @@ public KeyVaultEnvironmentPostProcessorHelper(final ConfigurableEnvironment envi
...
+    /**
+     * Add a key vault property source.
+     *
+     * <p>
+     * The normalizedName is used to target a specific key vault (note if the
+     * name is the empty string it works as before with only one key vault
+     * present).
+     * </p>
+     *
+     * @param postfix the postfix.
```
Comment (saragluna) [inline azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java:57]: inaccurate parameter name `postfix`

File: azure-spring-boot-tests/azure-spring-boot-test-keyvault/pom.xml [hunk 2]
```diff
@@ -15,6 +15,41 @@
...
+                                <requireEnvironmentVariable>
+                                    <variableName>AZURE_TENANT</variableName>
+                                    <message>Set the AZURE_TENANT environment variable to your Azure Tenant ID</message>
+                                </requireEnvironmentVariable>
+                                <requireEnvironmentVariable>
+                                    <variableName>AZURE_SUBSCRIPTION</variableName>
+                                    <message>Set the AZURE_SUBSCRIPTION environment variable to your Azure Subscription ID</message>
+                                </requireEnvironmentVariable>
+                                <requireEnvironmentVariable>
+                                    <variableName>AZURE_CLIENT_ID</variableName>
```
[the remark this answers is not in the document]
Comment (mnriem) [reply azure-spring-boot-tests/azure-spring-boot-test-keyvault/pom.xml:42]: Added it to the POM

File: azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java (hunk 1 above)
Comment (mnriem) [reply azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java]: Changed it to `normalizedName` and added a more detailed explanation of what the normalizedName is.

**Comment by mnriem** [comment]
When you say the document which one do you mean? 

On the order yes we need to clarify that the order of the .names attribute determines the order in which the respective KeyVaults are consulted. 

On the naming convention I opted for the key vault name at the beginning as it was easier/cleaner in the implementation to add it, but I have no preference. So if it needs to be changed just say the word.

**Comment by saragluna** [comment]
Sorry for not mentioning where to add them. A section describing the new feature added to the [README.md](https://github.com/microsoft/azure-spring-boot/blob/master/azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md) will be enough. 

If the `<keyvault-name>` has no usage other than distinguishing different configurations, we'd better use array to list these configurations, in which case the order could be expressed by their order showing up in the array. 

If we want to use the `<keyvault-name>` pattern,  I think `azure.keyvault.<keyvault-name>.xxx` is better even though the implementation isn't clean/compact.

**Comment by mnriem** [comment]
OK I will change it to use to the ```azure.keyvault.<keyvault-name>.xxx` syntax. I will rename the `.names` property to `.order` as it expresses what it is used for from the customers perspective, it is required however as it is also used to whether or not to enable multiple key vault support or not.

**Comment by saragluna** [comment]
There're some unit tests failed.

File: azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md [hunk 3]
```diff
@@ -96,4 +96,24 @@ azure.keyvault.allow.telemetry=false
 When telemetry is enabled, an HTTP request will be sent to URL `https://dc.services.visualstudio.com/v2/track`. So please make sure it's not blocked by your firewall.    
 Find more information about Azure Service Privacy Statement, please check [Microsoft Online Services Privacy Statement](https://www.microsoft.com/en-us/privacystatement/OnlineServices/Default.aspx). 
 
+## Multiple Key Vault support
```
Comment (saragluna) [inline azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md:99]: What if the user also sets the `azure.keyvault.uri`, what's the order between them?

Comment (mnriem) [reply azure-spring-boot-starters/azure-keyvault-secrets-spring-boot-starter/README.md:99]: The order would be dependent on the order in the file, but I feel we should caution a customer against doing that. If they opt for multiple key vaults they should only use it that way so the order is guaranteed. I will add this to the documentation.

**Pull request merged**